### PR TITLE
Fixed Dedupe plugin for Maple engine

### PR DESCRIPTION
**Pull request by andreychumak**
See details in https://github.com/webpack/webpack/pull/2625

### CODE REVIEW

File: lib/optimize/DedupePlugin.js [hunk 1]
```diff
@@ -207,6 +212,11 @@ DedupePlugin.prototype.apply = function(compiler) {
 							"}(modules[i]));",
 							"break;"
 						]),
+						"case \"undefined\":",
+						this.indent([
+							"// Module is skipped in this bundle",
+							"break;"
+						]),
 						"default:",
```
Comment (sokra) [inline lib/optimize/DedupePlugin.js]: Could you rewrite the `default` branch to `case "number":`? This should have the same effect, but with smaller size. Then we could also remove the `"function"` branch. everything should be ignored by default.

Comment (andreychumak) [inline lib/optimize/DedupePlugin.js]: @sokra , shouldn't it be `case "number":` instead of `case "string":`?

File: lib/optimize/DedupePlugin.js [hunk 2]
```diff
@@ -207,7 +206,7 @@ DedupePlugin.prototype.apply = function(compiler) {
 							"}(modules[i]));",
 							"break;"
 						]),
-						"default:",
+						"case \"number\":",
```
Comment (sokra) [inline lib/optimize/DedupePlugin.js]: String instead of number?

**Comment by SpaceK33z** [comment]
Closing since the DedupePlugin was removed for webpack v2.

**Pull request closed without merging**